tapiriik/services/Pulsstory/pulsstory.py

```
@@ -0,0 +1,385 @@
+                else:
+                    wpType = "gps"  # meh
+                '''
+
+                if not inPause and waypoint_type == WaypointType.Pause:
+                    inPause = True
+                elif inPause and waypoint_type == WaypointType.Pause:
+                    continue
+                elif inPause and waypoint_type != WaypointType.Pause:
+                    inPause = False
+
+                if waypoint.HR is not None:
+                    record["Waypoints"]["HeartRateValue"].append(round(waypoint.HR))
+
+                if waypoint.Power is not None:                    
+                    record["Waypoints"]["PowerValue"].append(waypoint.Power)
+                    
+                if waypoint.Speed is not None:                    
+                    record["Waypoints"]["SpeedValue"].append(waypoint.Speed)
+                    
+                if waypoint.Cadence is not None:                    
+                    record["Waypoints"]["CadanceValue"].append(waypoint.Cadence)
+                    
+                if waypoint.Location is not None and waypoint.Location.Longitude is not None:                    
+                    record["Waypoints"]["LongitudePathValue"].append(waypoint.Location.Longitude)
+                    
+                if waypoint.Location is not None and waypoint.Location.Latitude is not None:
+                    record["Waypoints"]["LatitudePathValue"].append(waypoint.Location.Latitude)
+                
+                if waypoint.Location is not None and waypoint.Location.Altitude is not None:                 
+                    record["Waypoints"]["AltitudePathValue"].append(waypoint.Location.Altitude)
+
+        return record
+
+    def DeleteCachedData(self, serviceRecord):
+        cachedb.rk_activity_cache.remove({"Owner": serviceRecord.ExternalID})
+
+    def DeleteActivity(self, serviceRecord, uri):
```

Leave implemented so an exception gets thrown if something were to attempt to call this, rather than failing silently.